pkg/virt-controller/watch/migration.go

```diff
@@ -501,88 +510,9 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
-				condition := virtv1.VirtualMachineInstanceMigrationCondition{
-					Type:          virtv1.VirtualMachineInstanceMigrationRejectedByResourceQuota,
-					Status:        k8sv1.ConditionTrue,
-					LastProbeTime: v1.Now(),
-				}
-				migrationCopy.Status.Conditions = append(migrationCopy.Status.Conditions, condition)
-			}
-		case virtv1.MigrationScheduling:
-			if conditionManager.HasCondition(migrationCopy, virtv1.VirtualMachineInstanceMigrationRejectedByResourceQuota) {
-				conditionManager.RemoveCondition(migrationCopy, virtv1.VirtualMachineInstanceMigrationRejectedByResourceQuota)
-			}
-			if isPodReady(pod) {
-				if controller.VMIHasHotplugVolumes(vmi) {
-					if attachmentPodExists && isPodReady(attachmentPod) {
-						log.Log.Object(migration).Infof("Attachment pod %s for vmi %s/%s is ready", attachmentPod.Name, vmi.Namespace, vmi.Name)
-						migrationCopy.Status.Phase = virtv1.MigrationScheduled
-					}
-				} else {
-					migrationCopy.Status.Phase = virtv1.MigrationScheduled
-				}
-			}
-		case virtv1.MigrationScheduled:
-			if vmi.Status.MigrationState != nil &&
-				vmi.Status.MigrationState.MigrationUID == migration.UID &&
-				vmi.Status.MigrationState.TargetNode != "" {
-				migrationCopy.Status.Phase = virtv1.MigrationPreparingTarget
-			}
-		case virtv1.MigrationPreparingTarget:
-			if vmi.Status.MigrationState.TargetNode != "" && vmi.Status.MigrationState.TargetNodeAddress != "" {
-				migrationCopy.Status.Phase = virtv1.MigrationTargetReady
-			}
-		case virtv1.MigrationTargetReady:
-			if vmi.Status.MigrationState.StartTimestamp != nil {
-				migrationCopy.Status.Phase = virtv1.MigrationRunning
-			}
-		case virtv1.MigrationRunning:
-			_, exists := pod.Annotations[virtv1.MigrationTargetReadyTimestamp]
-			if !exists && vmi.Status.MigrationState.TargetNodeDomainReadyTimestamp != nil {
-				key := patch.EscapeJSONPointer(virtv1.MigrationTargetReadyTimestamp)
-				patchOps := fmt.Sprintf(`[{ "op": "add", "path": "/metadata/annotations/%s", "value": "%s" }]`,
-					key,
-					vmi.Status.MigrationState.TargetNodeDomainReadyTimestamp.String())
-
-				_, err := c.clientset.CoreV1().Pods(pod.Namespace).Patch(context.Background(), pod.Name, types.JSONPatchType, []byte(patchOps), v1.PatchOptions{})
-				if err != nil {
-					return err
-				}
-			}
-
-			if vmi.Status.MigrationState.Completed &&
-				!vmiConditionManager.HasCondition(vmi, virtv1.VirtualMachineInstanceVCPUChange) {
-				migrationCopy.Status.Phase = virtv1.MigrationSucceeded
-				c.recorder.Eventf(migration, k8sv1.EventTypeNormal, SuccessfulMigrationReason, "Source node reported migration succeeded")
-				log.Log.Object(migration).Infof("VMI reported migration succeeded.")
-			}
+		err := c.processMigrationPhase(migrationPhaseArgs, migration, migrationCopy, vmi, syncError)
```

Comment:
Yeah, personally I was not really sure whether in this case it is better to have a long function signature with each parameter listed out or to have a struct to contain the parameters. I know that [sonar cloud](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&id=kubevirt_kubevirt) flags functions with more than 7 parameters so that is why I chose to use the struct based on what sonar cloud was flagging as code smell.